content/articles/momentum-based-gradient-optimizer-and-optimization-techniques-for-gradient-descent/index.md

```diff
@@ -0,0 +1,193 @@
+   alt: Gradient Descent Based Optimization Techniques example image
+---
+Gradient descent is used to optimize neural networks and machine learning algorithms. This post delves into the inner workings of some of the most popular gradient-based optimization algorithms. These algorithms include Momentum, Adagrad, and Adam.
+
+### Table of contents
+- [Table of contents](#table-of-contents)
+- [Optimization techniques](#optimization-techniques)
+  - [Why do we optimize our machine learning models?](#why-do-we-optimize-our-machine-learning-models)
+- [Momentum-based optimization](#momentum-based-optimization)
+- [How does momentum help in gradient descent?](#how-does-momentum-help-in-gradient-descent)
+- [Stochastic Gradient Descent](#stochastic-gradient-descent)
+- [Adaptive Moment Estimation](#adaptive-moment-estimation)
+- [RMSProp](#rmsprop)
+- [AdaGrad](#adagrad)
+- [AdaDelta](#adadelta)
+  - [There have been significant developments on the Adadelta front](#there-have-been-significant-developments-on-the-adadelta-front)
+- [Mini Batch Stochastic Gradient Descent](#mini-batch-stochastic-gradient-descent)
+- [Various other optimizers that have emerged in the last few years](#various-other-optimizers-that-have-emerged-in-the-last-few-years)
+- [How to choose an optimizer](#how-to-choose-an-optimizer)
+  - [Optimizer to use](#optimizer-to-use)
+- [Conclusion](#conclusion)
+
+### Optimization techniques
+We will look at various optimization methods frequently employed in deep learning. We will also look at some Adaptive Optimization Algorithms (AOA). Compiling a program based on the current execution profile is known as adaptive optimization in computer science. You can easily make a tradeoff between Just-In-Time (JIT) compilation and instruction interpretation in an adaptive optimizer solution.
+
+#### Why do we optimize our machine learning models?
+Every iteration is compared to the previous one, and the hyper-parameters are tweaked until the results are as good as they can be. We can produce a model that is both accurate and error-free. 
+
+This page includes a quick overview of the many approaches that are accessible, including those provided by the Keras package.
+
+### Momentum-based optimization
+Momentum-based optimization utilizes an Adaptive Optimization Algorithm (AOA) that uses exponentially-weighted averaging gradients from prior rounds to keep convergence from deviating. Unfortunately, most real-world applications such as deep neural networks rely on noisy input to train.
+
+There is a systematic approach to feeding the data to minimize the impact of noise during optimization. For example, using exponentially weighted averages can solve.
+
+Using an iterative process, we can approximate the trends in a P-dimensional noisy dataset by calculating the parameters as we go through the data:
+
+$$
+On\; iteration\; p\\

```

Comment:
The new line character isn't working. Can you check that? We can move to final review once this is rectified.